src/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -240,5 +240,82 @@ private class SerializableLocalToolSinglePackage
             public string version { get; set; }
             public string[] commands { get; set; }
         }
+
+        public FilePath FindFirst()
+        {
+            foreach ((FilePath possibleManifest, DirectoryPath _) in EnumerateDefaultAllPossibleManifests())
+            {
+                if (_fileSystem.File.Exists(possibleManifest.Value))
+                {
+                    return possibleManifest;
+                }
+            }
+
+            throw new ToolManifestCannotBeFoundException(
+                    string.Format(LocalizableStrings.CannotFindAnyManifestsFileSearched,
+                        string.Join(Environment.NewLine, EnumerateDefaultAllPossibleManifests().Select(f => f.manifestfile.Value))));
+        }
+
+        public void Add(
```

Comment:
Seems weird that the Finder is also responsible for the Add. Shouldn't this be in the Manifest itself?

Reply:
it get renamed in the next commit. the whole class is called ToolManifestFile now. it is hard to separate find and read since depends on the content we need to keep looking

Reply:
separate ToolManifestFinder and ToolManifestEditor